logdetective/utils.py

```diff
@@ -57,20 +58,32 @@ def initialize_model(model_pth: str, filename_suffix: str = ".gguf", verbose: bo
         model = Llama(
             model_path=model_pth,
             n_ctx=0,  # Maximum context for the model
-            verbose=verbose)
+            verbose=verbose,
+            logits_all=True)
     else:
         model = Llama.from_pretrained(
             model_pth,
             f"*{filename_suffix}",
             n_ctx=0,  # Maximum context for the model
-            verbose=verbose)
+            verbose=verbose,
+            logits_all=True)
 
     return model
 
 
-def process_log(log: str, model: Llama, stream: bool) -> str:
+def compute_certainty(probs: List[Dict[str, float] | None]) -> float:
+    """Compute certainty of repsponse based on average logit probability
     """
-    Processes a given log using the provided language model and returns its summary.
+
+    top_logprobs = [
+        np.exp(x) * 100 for e in probs if isinstance(e, dict) for x in e.values()]
```

Comment:
I will welcome comment on why this formula.

Reply:
Log probability is `log(p)`, isn't really readable for most people, especially in compound. IT needs to be converted first. 

In this case it's just a  matter of applying inverse operation `exp`. Of course that leaves you with a value in range `<0, 1>` so it needs to be multiplied by `100`.

Reply:
Can you put these two lines as comment there. It will be helpful for any of us fixing this code 5 years in the future.